clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -621,7 +621,8 @@ private void doCommitOffsetsAsync(final Map<TopicPartition, OffsetAndMetadata> o
             public void onSuccess(Void value) {
                 if (interceptors != null)
                     interceptors.onCommit(offsets);
-
+                offsets.forEach((tp, offsetMeta) ->
```

Comment:
Another case we need to use `maybeRequestUpdate` is when we are loading the committed offsets for the consumer. This is an interesting case because we will not necessarily have any metadata corresponding to the loaded epoch. It is possible that when we refresh metadata, we will see an older epoch, which will cause us to ignore the update for that partition. In that case, we probably need some way to signal the need to retry. Not sure it's a good idea, but one possibility might be to replace the `needUpdate` flag with a set of the partitions that need to be updated.

Reply:
For now, I added `updateLastSeenEpoch` calls in the three "seek" methods in KafkaConsumer